draft-ietf-tls-tls13.md

```diff
@@ -2293,14 +2295,24 @@ Clients MUST NOT use cookies in their initial ClientHello in subsequent connecti
 
 ###  Signature Algorithms
 
-The client uses the "signature_algorithms" extension to indicate to the server
-which signature algorithms may be used in digital signatures. Clients which
-desire the server to authenticate itself via a certificate MUST send this extension.
-If a server is authenticating via a certificate and the client has not sent a
-"signature_algorithms" extension, then the server MUST
-abort the handshake with a "missing_extension" alert (see {{mti-extensions}}).
-
-The "extension_data" field of this extension in a ClientHello contains a
+TLS 1.3 provides two extensions for indicating which signature
+algorithms may be used in digital signatures. The
+"signature_algorithms_cert" extension applies to signatures in
+certificates and to the and the "signature_algorithms" extension, which originally
+appeared in TLS 1.2, applies signatures in CertificateVerify
+messages. The keys found in certificates MUST also be of
+appropriate type for the signature algorithms they are used
+with. This is a particular issue for RSA keys and PSS signatures,
+as described below. If no "signature_algorithms_cert" extension is present,
+then the "signature_algorithms" extension also applies to signatures
+appearing in certificates. Clients which desire the server to authenticate
+itself via a certificate MUST send "signature_algorithms". If a server
+is authenticating via a certificate and the client has not sent a
+"signature_algorithms" extension, then the server MUST abort the
+handshake with a "missing_extension" alert (see {{mti-extensions}}).
+TLS 1.2 implementations SHOULD also process this extension.
```

Comment:
That's a large deviation from the -21 draft, for the benefit of specific platforms. I don't believe that text should be merged in TLS1.3 as is. The requirement:
```
If a server
+is authenticating via a certificate and the client has not sent a
+"signature_algorithms" extension, then the server MUST abort the
+handshake with a "missing_extension" alert (see {{mti-extensions}}).
```
is punishing clients which do not need to send this extension, i.e., clients which their validator has no issue handling RSA-PSS. Why this isn't an optional extension?

Reply:
This requirement has been in TLS 1.3 for some time. See:
https://tlswg.github.io/tls13-spec/draft-ietf-tls-tls13.html#rfc.section.4.2.3

In fact, it's required to be there because the default "signature_algorithms" value (from 5246) does not indicate that you support SHA-256.